tagging.py

```
@@ -1,16 +1,503 @@
+    
+    def run(self) -> None:
+        # Keeps the lexer process running
+        self.state = run_lexer
+        while self.state is not None:
+            self.state = self.state(self)
+
+
+def errorf(lex: Lexer, message: str) -> None:
+    lex.items.append(Item(ItemType.Error, lex.start, message))
+
+def run_lexer( lex: Lexer) -> LexerFunc:
```

Ruff (0.11.9)

`run_lexer` is too complex (20 > 10)

(C901)